python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py

```diff
@@ -2494,27 +2495,45 @@ def saveMetric(
         is Free Energy protocol, the dHdl and the u_nk data will be saved in the
         same parquet format as well.
         """
-        datadict = {
-            "Time (ps)": [time / _Units.Time.picosecond for time in self.getTime(True)],
-            "PotentialEnergy (kJ/mol)": [
+        datadict = dict()
+        if isinstance(self._protocol, _Protocol.Minimisation):
+            datadict["Time (ps)"] = self.getStep(True)
+            datadict["PotentialEnergy (kJ/mol)"] = [
+                energy / _Units.Energy.kj_per_mol
+                for energy in self.getTotalEnergy(True)
+            ]
+        else:
+            datadict["Time (ps)"] = [
+                time / _Units.Time.picosecond for time in self.getTime(True)
+            ]
+            datadict["PotentialEnergy (kJ/mol)"] = [
                 energy / _Units.Energy.kj_per_mol
                 for energy in self.getPotentialEnergy(True)
-            ],
-            "Volume (nm^3)": [
-                volume / _Units.Volume.nanometer3 for volume in self.getVolume(True)
-            ],
-            "Pressure (bar)": [
+            ]
+            if self.getVolume():
+                datadict["Volume (nm^3)"] = [
+                    volume / _Units.Volume.nanometer3 for volume in self.getVolume(True)
+                ]
+            datadict["Pressure (bar)"] = [
                 pressure / _Units.Pressure.bar for pressure in self.getPressure(True)
-            ],
-            "Temperature (kelvin)": [
+            ]
+            datadict["Temperature (kelvin)"] = [
                 temperature / _Units.Temperature.kelvin
                 for temperature in self.getTemperature(True)
-            ],
-        }
-        df = pd.DataFrame(data=datadict)
+            ]
+
+        try:
+            df = pd.DataFrame(data=datadict)
+        except ValueError:
+            length_dict = {key: len(value) for key, value in datadict.items()}
+            warnings.warn(f'Not all metric has the same number of data points ({length_dict}).'
+                          f'All columns will be truncated the same length.')
+            length = min(length_dict.values())
+            new_datadict = {key: value[:length] for key, value in datadict.items()}
+            df = pd.DataFrame(data=new_datadict)
         df = df.set_index("Time (ps)")
         df.to_parquet(path=f"{self.workDir()}/{filename}", index=True)
-        if isinstance(self._protocol, _Protocol._FreeEnergyMixin):
+        if isinstance(self._protocol, _Protocol.FreeEnergy):
```

Comment:
Why is that changed?

Reply:
Because neither FreeEnergyEquilibration nor FreeEnergyMinimisation generate sensible energy files